shell/run_shell_cmd.go

```diff
@@ -66,16 +65,26 @@ func RunShellCommandWithOutput(terragruntOptions *options.TerragruntOptions, wor
 		cmd.Dir = workingDir
 	}
 	// Inspired by https://blog.kowalczyk.info/article/wOYk/advanced-command-execution-in-go-with-osexec.html
-	cmd.Stderr = io.MultiWriter(errWriter, &stderrBuf)
+	cmdStderr := io.MultiWriter(errWriter, &stderrBuf)
+	var cmdStdout io.Writer
 	if !suppressStdout {
-		cmd.Stdout = io.MultiWriter(outWriter, &stdoutBuf)
+		cmdStdout = io.MultiWriter(outWriter, &stdoutBuf)
 	} else {
-		cmd.Stdout = io.MultiWriter(&stdoutBuf)
+		cmdStdout = io.MultiWriter(&stdoutBuf)
 	}
 
-	if err := cmd.Start(); err != nil {
-		// bad path, binary not executable, &c
-		return nil, errors.WithStackTrace(err)
+	if terragruntOptions.AllocatePseudoTTY {
+		if err := runCommandWithPTTY(terragruntOptions, cmd, cmdStdout, cmdStderr); err != nil {
```

Comment:
Sanity check: any reason not to _always_ run with a pseudo TTY so we don't need an extra flag to document, maintain, etc? 

Or, alternatively, IIRC, the `terminal` package in Go can tell you if you're currently in a terminal / tty, and we can use that to decide this automatically.

Reply:
It isn't actually whether or not we are in a terminal, but rather whether or not the command we want to run is interactive so that we get proper prompts and readline support when we call it via `exec`. So really, the question is whether the command we are about to run is interactive or not. I guess we can maintain a list of terraform commands that are interactive, since it is unlikely terraform will have another command like `console`.

Will go ahead and update that logic.

Reply:
> Sanity check: any reason not to always run with a pseudo TTY so we don't need an extra flag to document, maintain, etc?

Also, we don't want to do this all the time, because it messes up stderr from the commands.

Reply:
UPDATE (2704efe): Switched to automatically use ptty routine if the terraform command is `console`.

Reply:
Roger, thx!